salt/cloud/clouds/vmware.py

```diff
@@ -994,32 +1054,36 @@ def _wait_for_ip(vm_ref, max_wait):
             return False
     time_counter = 0
     starttime = time.time()
+    ipv4_address = None
     while time_counter < max_wait_ip:
         if time_counter % 5 == 0:
             log.info(
-                "[ %s ] Waiting to retrieve IPv4 information [%s s]",
+                "[ %s ] Waiting to retrieve IPv4/6 information [%s s]",
                 vm_ref.name, time_counter
             )
 
-        if vm_ref.summary.guest.ipAddress and _valid_ip(vm_ref.summary.guest.ipAddress):
-            log.info(
-                "[ %s ] Successfully retrieved IPv4 information in %s seconds",
-                vm_ref.name, time_counter
-            )
-            return vm_ref.summary.guest.ipAddress
         for net in vm_ref.guest.net:
             if net.ipConfig.ipAddress:
                 for current_ip in net.ipConfig.ipAddress:
-                    if _valid_ip(current_ip.ipAddress):
+                    if _valid_ip6(current_ip.ipAddress):
```

Comment:
Most software nowadays use IPv6 when both v4 and v6 is available. I believe doing the exact opposite is not something we should do.

Minor flaw: this will break if the master does not have an IPv6 addr while the client does have one. Is there a check available in the codebase to determine if the master has IPv6 connectivity?